bundle/edu.gemini.spdb.reports.collection/src/main/java/edu/gemini/dbTools/odbState/ObservationState.java

```diff
@@ -144,42 +143,39 @@ public TooType getTooType() {
     }
 
     /**
-     * Gets a formated string that indicates the night during which the
+     * Gets a formatted string that indicates the night during which the
      * observation took place.
      * @param time utc time of the last observation event message
      * @return formatted string indicating the night during which the
      * observation took place.
      */
     private static String _getNight(final long time) {
-        final Calendar cal = GregorianCalendar.getInstance();
-        cal.setTimeInMillis(time);
-
-        // The calendar is in the local time zone.  If the time is before
-        // noon, then the night started on the previous day.  If the time
-        // is after noon, then the night ends on the next day.
-        final int ampm = cal.get(Calendar.AM_PM);
-
-        final Calendar endCal;
-
-        final int startDay;
-        final int endDay;
-        if (ampm == Calendar.AM) {
-            endCal = (Calendar) cal.clone();
-            endDay = cal.get(Calendar.DAY_OF_MONTH);
-            cal.add(Calendar.DAY_OF_MONTH, -1);
-            startDay = cal.get(Calendar.DAY_OF_MONTH);
+        // Determine when, in the local time zone, the time started:
+        // If the time is before noon, then the night started on the previous day.
+        // If the time is after noon, then the night ends on the next day.
+        // Note: AM is 0, PM is 1 by documentation for ChronoField.
+        final LocalDateTime dt = LocalDateTime.ofInstant(Instant.ofEpochMilli(time), ZoneId.systemDefault());
+        final int ampm = dt.get(ChronoField.AMPM_OF_DAY);
+
+        final LocalDateTime start;
+        final LocalDateTime end;
+        if (ampm == 0) {
+            start = dt.minus(1, ChronoUnit.DAYS);
+            end   = dt;
         } else {
-            startDay = cal.get(Calendar.DAY_OF_MONTH);
-            cal.add(Calendar.DAY_OF_MONTH, 1);
-            endCal = cal;
-            endDay = cal.get(Calendar.DAY_OF_MONTH);
+            start = dt;
+            end   = dt.plus(1, ChronoUnit.DAYS);
         }
 
+        final int startDay = start.get(ChronoField.DAY_OF_MONTH);
+        final int endDay   = end.get(ChronoField.DAY_OF_MONTH);
+
         final StringBuilder buf = new StringBuilder();
         buf.append(startDay).append("/").append(endDay).append(" ");
 
-        final DateFormat f = new SimpleDateFormat("MMM yyyy");
-        buf.append(f.format(endCal.getTime()));
+        // Do not need to explicitly set a ZoneId since we are only accessing MMM and yyyy.
+        final DateTimeFormatter f = DateTimeFormatter.ofPattern("MMM yyyy");
+        buf.append(f.format(end));
```

Comment:
> Investigating further, it seems like in the original case (as well as the new case whether I specify with ZoneId.of("UTC") or nothing) the time is being printed in UTC.

I think this version of the code with the new time API is better than the previous one but I don't understand why it would format the date in UTC.  It appears to me that both cases should be in local time.  For Gemini South that is going to correspond to the same "night" as the time in UTC given that we finish observing before noon UTC.  At Gemini North, I'm not so sure.

At any rate, it doesn't appear that this formatted `String` or the corresponding night is ever used by anything (the `_night` field is private and there is no accessor) so it probably doesn't matter.

Reply:
I was a bit confused by how this was supposed to work and why we need a night in the first place, but it seemed "wonky" to me in its original format, so I just tried to modernize without changing the functionality, since it didn't seem logical to me.

It does seem like perhaps this can all just be removed, but as we're six days from release, I don't want to muck something unexpected up by doing so.